### ISSUE TITLE
‘Reflection’ in ‘class google::protobuf::Message’ does not name a type...

### ISSUE THREAD

**Issue by ecmagalhaes** [protobuf-c/protobuf-c#363]
Hi, I tried to make protobuf-c, using the latest protobuf repo and encountered the same error.

It seems to be solved before however still got the same error.

**Comment by ecmagalhaes** [comment]
```
t/generated-code2/cxx-generate-packed-data.cc: In function ‘void dump_test_unknown_fields()’:
t/generated-code2/cxx-generate-packed-data.cc:1001:36: error: ‘Reflection’ in ‘class google::protobuf::Message’ does not name a type
   const google::protobuf::Message::Reflection *reflection = mess.GetReflection(
                                    ^
t/generated-code2/cxx-generate-packed-data.cc:1002:43: error: ‘reflection’ was not declared in this scope
   google::protobuf::UnknownFieldSet *fs = reflection->MutableUnknownFields(&mes
                                           ^
Makefile:1524: recipe for target 't/generated-code2/t_generated_code2_cxx_generate_packed_data-cxx-generate-packed-data.o' failed
make: *** [t/generated-code2/t_generated_code2_cxx_generate_packed_data-cxx-generate-packed-data.o] Error 1
```

**Comment by ecmagalhaes** [comment]
- Ubuntu 16.04
- protobuf-c 1.3.1
- libprotoc 3.7.0

**Comment by Mabsss83** [comment]
Same here
```t/generated-code2/cxx-generate-packed-data.cc: In function ‘void dump_test_unknown_fields()’:
t/generated-code2/cxx-generate-packed-data.cc:1001:36: error: ‘Reflection’ in ‘class google::protobuf::Message’ does not name a type
   const google::protobuf::Message::Reflection *reflection = mess.GetReflection();
                                    ^~~~~~~~~~
t/generated-code2/cxx-generate-packed-data.cc:1002:43: error: ‘reflection’ was not declared in this scope
   google::protobuf::UnknownFieldSet *fs = reflection->MutableUnknownFields(&mess);
                                           ^~~~~~~~~~
t/generated-code2/cxx-generate-packed-data.cc:1002:43: note: suggested alternative: ‘sigaction’
   google::protobuf::UnknownFieldSet *fs = reflection->MutableUnknownFields(&mess);
                                           ^~~~~~~~~~
                                           sigaction
make: *** [Makefile:1558: t/generated-code2/cxx_generate_packed_data-cxx-generate-packed-data.o] Error 1
```

- ubuntu 18.10
- protobuf-c 1.3.1
- libprotoc 3.0.0-9.1ubuntu3

(just copied the config info above, not knowing at all what relevent things to put)

**Comment by Mabsss83** [comment]
Modifying the :
`
const google::protobuf::Message::Reflection *reflection = mess.GetReflection();
`

to :
`const google::protobuf::Reflection *reflection = mess.GetReflection();`

in 
/protobuf-c-master/t/generated-code2/cxx-generate-packed-data.cc 
at line 1001 worked for compiling for me.

Now I am stuck with an install error.

**Comment by JoowonYun** [comment]
I already fixed in #342 .

You can use this pull-request.
